hummingbot/hummingbot
# fix/order tracker not match orders with undefined ids

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
PR created to solve the issue https://github.com/hummingbot/hummingbot/issues/5056
Change the fetch order logic in the client order tracker to not look for orders using the exchange order id if it has not been provided as a parameter. Without this change if the exchange order id is not specified it will be None, and the fetch method will match orders in the tracker without exchange order id.
Also in this PR the update logic from a trade update has been changed to never update the exchange order id. That id should be only updated by the order update.


**Tests performed by the developer**:
Added unit tests for the described scenarios.


**Test Summary**:
- Setup multiple test bots using a single Binance account (same API key)
- 3 bots running using PMM strategy and the rest are running Avellaneda MM and Liquidity mining strategy
- Let the test bots run for almost 24 hours
- Regularly check the filled records by running history --verbose and by browsing the TradeFill table 
- Verify amounts seen in the table that differ from the specified order amount in the strategy and confirmed that these are partial fills


**Tips for QA testing**:
It is not easy to replicate the scenario running the client. The best approach is to have several client instances running using the a Binance connector with the same user account (it could be for different trading pairs in each instance).


Fixes #5056 

This PR comes from https://github.com/CoinAlpha/hummingbot/pull/70
